tests/api/api/test_frontend_spec.py

```diff
@@ -131,3 +135,41 @@ def test_get_frontend_spec_jobs_dashboard_url_resolution(
         f"&var-groupBy={{filter_name}}&var-filter={{filter_value}}"
     )
     mlrun.api.utils.clients.iguazio.Client().try_get_grafana_service_url.assert_called_once()
+
+
+def test_get_frontend_spec_nuclio_streams(
+    db: sqlalchemy.orm.Session, client: fastapi.testclient.TestClient
+) -> None:
+    for testcase in [
+        {
+            "iguazio_version": "3.2.0",
+            "nuclio_version": "1.6.23",
+            "expected_feature_flag": mlrun.api.schemas.NuclioStreamsFeatureFlag.disabled,
+        },
+        {
+            "iguazio_version": None,
+            "nuclio_version": "1.6.23",
+            "expected_feature_flag": mlrun.api.schemas.NuclioStreamsFeatureFlag.disabled,
+        },
+        {
+            "iguazio_version": None,
+            "nuclio_version": "1.7.8",
+            "expected_feature_flag": mlrun.api.schemas.NuclioStreamsFeatureFlag.disabled,
+        },
+        {
+            "iguazio_version": "3.4.0",
+            "nuclio_version": "1.7.8",
+            "expected_feature_flag": mlrun.api.schemas.NuclioStreamsFeatureFlag.enabled,
+        },
+    ]:
+        # init cached value to None in the beginning of each test case
+        mlrun.runtimes.utils.cached_nuclio_version = None
+        mlrun.mlconf.igz_version = testcase["iguazio_version"]
```

Comment:
```suggestion
        mlrun.mlconf.igz_version = test_case.get("iguazio_version")
```
Using `test_case["x"]` requires you to define x for all tests, otherwise, the tests will fail because the attribute does not exist. Another possible issue is that if you add a new test that uses a new attribute, you must also add this attribute to all existing tests
When using `test_case.get("x")` if `x` does not exist in a specific test, the default value is used, this allows us to avoid updating older test cases and better understand what each test case tests.